Added more helpful message for AmbiguousParameterSet exception

kvprasoon opened the pull request:
Had issues with my old fork and it got deleted, this issue is to close #5435

TravisEz13 committed:
update version regex for nanoserver docker (#5494)

TravisEz13 committed:
Fix mac packaging (#5496)

TravisEz13 committed:
Fix mac packaging (#5496)

kvprasoon committed:
Updated the error message as suggested .

markekraus reviewed (approved):
LGTM

SteveL-MSFT commented on src/System.Management.Automation/resources/ParameterBinderStrings.resx:
```diff
@@ -142,7 +142,7 @@
   </data>
   <data name="AmbiguousParameterSet" xml:space="preserve">
-    <value>Parameter set cannot be resolved using the specified named parameters.</value>
+    <value>Parameter set cannot be resolved using the specified named parameters. One or more parameters issued cannot be used together or an insufficient number of parameters were provided.</value>
```
Not a grammar expert, but the last part of this sentence reads weird to me.  I know that `were` is used with plural nouns, so you would think that because `parameters` is used, it should be `were`, but I think the noun here is actually `an insufficient number of parameters` which is singular (the `an` gives it away) sort of like `...or an error was provided.`.  So it seems that it would be `...parameters was provided.`.

This is not blocking feedback.

markekraus replied:
@SteveL-MSFT This is one of those weird English rules. 'a number of X-es` are plural while `the number of X-es` is singular. In this case we are using `an insufficient number of parameters` and is the plural form of the phrase. https://en.oxforddictionaries.com/explore/number-of-people-is-or-are

iSazonov replied:
:-) Now I know who to ask to approve error messages.

iSazonov replied:
@markekraus Could you please look test failures? It seems HttpListener can not start on 8082 port  in `Describing Web cmdlets tests using the cmdlet's aliases`. Also these tests is "alias" tests - I think we shiould remove its (we have separate test set for aliases).

markekraus replied:
@iSazonov There is an open issue for tracking the HttpListener issue #5464.

bergmeister committed:
Change synopsis of install-powershell.ps1 to reflect that it works cross-platform (#5465)

bergmeister committed:
Change synopsis of install-powershell.ps1 to reflect that it works cross-platform (#5465)

markekraus committed:
[Feature] Fix AppVeyor Fails (#5520)

TravisEz13 committed:
Update target tag for master CI to 6.1.0-preview.1 (#5513)

* use metadata JSON to create release tag in CI

TravisEz13 committed:
Update target tag for master CI to 6.1.0-preview.1 (#5513)

* use metadata JSON to create release tag in CI

TravisEz13 committed:
Fix docker tests (#5508)

* update expected version
* update container list

TravisEz13 committed:
Fix docker tests (#5508)

* update expected version
* update container list

TravisEz13 committed:
Replace strlen with strnlen in native code (#5510)

* fix expected vs actual for easier troubleshooting
* replace strlen calls with strnlen

iSazonov commented:
@kvprasoon Please rebase to get latest updates and pass CIs.

kvprasoon reviewed (commented):
rebased as appveyor CI fix went in to the base

markekraus committed:
Fix Import-CliXml Tests (#5521)

Add checks that ProcessName is not null or empty.

bergmeister committed:
Disambiguate icon for daily builds on Windows (#5467)

* Disambiguate icon for daily builds on Windows.
Some code had to be borrowed from build.psm1 because this script has to be self contained in case it gets executed by only downloading this file via the published download link https://twitter.com/Steve_MSFT/status/930585082451992576

kvprasoon commented:
I am not using cmdline, any  idea on how to do it via WEB UI ?

iSazonov commented:
@kvprasoon It seems Web UI does not have the capability.
If you plan to continue work with GitHub best start is to install [GitHub Desktop](https://help.github.com/desktop/guides/getting-started-with-github-desktop/installing-github-desktop/) application. It automatically configure an environment for you. Then you can run Git bash from it and make rebase from command line.

kvprasoon committed:
Revert "Change synopsis of install-powershell.ps1 to reflect that it works cross-platform (#5465)"

This reverts commit 59a78d54cc98235ab9417807d50da51b83a87cb2.

kvprasoon committed:
Revert "Fix mac packaging (#5496)"

This reverts commit 0aeec3c27fd4342423c8e994f6083f7e559334d5.

kvprasoon committed:
Revert "Update target tag for master CI to 6.1.0-preview.1 (#5513)"

This reverts commit f82438477d8f11c9fdc3cc8aa3488f4ee8083ead.

kvprasoon committed:
Revert "Fix docker tests (#5508)"

This reverts commit ea86a7d3ae5dfb95f9683c34f2d7f7e3af11d61e.

kvprasoon commented:
@iSazonov  oops, messed up something...

iSazonov commented:
@kvprasoon You need to learn Git docs and samples. I started out a year ago like you.

iSazonov commented:
@kvprasoon We can not merge the PR until you rebase.

kvprasoon commented:
Sorry @iSazonov , my fork is completely messed up, I would re-create and submit.

kvprasoon commented:
This will be addressed by [https://github.com/PowerShell/PowerShell/pull/5537](url)
